tests/ga/test_extension.py

```diff
@@ -1422,23 +1411,9 @@ def test__extension_upgrade_failure_when_prev_version_disable_fails_and_recovers
     @patch('azurelinuxagent.ga.exthandlers.HandlerManifest.get_disable_command')
     def test__extension_upgrade_failure_when_prev_version_disable_fails_incorrect_zip(self, patch_get_disable_command,
                                                                                       *args):
-        test_data = WireProtocolData(DATA_FILE_EXT_SINGLE)
-        exthandlers_handler, protocol = self._create_mock(test_data, *args)
-
-        # Ensure initial install and enable is successful
-        exthandlers_handler.run()
-        self.assertEqual(0, patch_get_disable_command.call_count)
-
-        self._assert_handler_status(protocol.report_vm_status, "Ready", expected_ext_count=1, version="1.0.0")
-        self._assert_ext_status(protocol.report_ext_status, "success", 0)
-
-        # Next incarnation, update version
-        test_data.goal_state = test_data.goal_state.replace("<Incarnation>1<", "<Incarnation>2<")
-        test_data.ext_conf = test_data.ext_conf.replace('version="1.0.0"', 'version="1.0.1"')
-        test_data.manifest = test_data.manifest.replace('1.0.0', '1.0.1')
+        test_data, exthandlers_handler, protocol = self._setup_update_test_and_update_gs(patch_get_disable_command,
```

Comment:
The tests look much cleaner with this being refactored in its own method. Nice!

Reply:
Thanks 💃